Add bookdown docs and restructure main README.md file.

shochdoerfer opened the pull request:
Introducing bookdown as a new dependency and moved the documentation from the main README.md file to the bookdown documentation.

shochdoerfer committed:
Add bookdown docs and restructure main README.md file.

settermjd committed:
Merge branch 'master' into feature/bookdown_docs

settermjd commented on README.md:
```diff
@@ -8,96 +8,56 @@ annotation-based dependency injection container. Have a look at the [disco-demos
-define the `scope` (which is either the string `request` or `session`), define a boolean value for `lazy`, or define
-a boolean value for `singleton`. The following configuration creates a lazy-loaded, singleton, request-aware instance:
+Next up you need to create a configuration class `MyConfiguration`. The
+class needs to be marked with a @Configuration annotation.
```
Would be good to markup `@Configuration`, to make it that much easier to read.

settermjd commented on README.md:
````diff
@@ -8,96 +8,56 @@ annotation-based dependency injection container. Have a look at the [disco-demos
 ```
 
-To inject a dependency, simply call the respective method:
+To declare an configuration entry, simply add a method to your
````
Should be _"a configuration entity"_ instead of _"an configuration entity"_. [Here's the rule](http://www.quickanddirtytips.com/education/grammar/a-versus-an) I base that statement on.

settermjd committed:
Update the README

This commit contains several changes, covering grammar, markup,
spelling, and some language usage conventions.

settermjd commented:
I made a couple of amendments. Thoughts?
